src/mca/ras/base/ras_base_allocate.c

```diff
@@ -189,17 +190,10 @@ void prte_ras_base_allocate(int fd, short args, void *cbdata)
                            PRTE_ATTR_LOCAL, NULL, PMIX_BOOL);
     }
 
-    /* if we already did this, don't do it again - the pool of
-     * global resources is set.
+    /* We may need to go through this patch multiple times, one for each job
+     * since some RAS components (e.g., LSF) define the mapper to be used for
+     * this specific job.
      */
-    if (prte_ras_base.allocation_read) {
```

Comment:
Could you please explain why you are removing this protection? I believe it is actually required by other components. Perhaps the solution is that LSF should not be going thru the ras to define the mapper? That seems an odd breakage of the abstraction.

Reply:
Each `jdata` is passed through the `prte_ras_base_allocate` routine. The first `jdata` is namespace 0 (for the daemons) so the RAS runs and defines the hosts. For LSF it also defines the mapper to be used along with the rankfile, and adds that information to the `jdata`. Now the daemons don't really care about the mapper, just the node list so no big deal.

The problem is when on the second call to `prte_ras_base_allocate` the `jdata` is for namespace 1 (for the application) which does need the information that LSF would store on the `jdata`. With the `prte_ras_base.allocation_read` in place then the LSF RAS is not called for the application job structure, thus the mapper is never set.

By removing the check then we allow the RAS to adjust the jdata per job, and influence the application job mapping.